doc/api/index.rst

```diff
@@ -172,6 +172,7 @@ Input/output
     :toctree: generated
 
     load_dataarray
+    read
```

Comment:
The `load_dataarray` function was put under the `pygmt.io` namespace. Should we consider putting `read` under `pygmt.io` too? (Thinking about whether we need a low-level `pygmt.clib.read` and high-level `pygmt.io.read` in my other comment).